gravitee-apim-gateway/gravitee-apim-gateway-flow/src/test/java/io/gravitee/gateway/jupiter/flow/condition/evaluation/PathBasedConditionFilterTest.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.jupiter.flow.condition.evaluation;
+
+import static org.mockito.Mockito.when;
+
+import io.gravitee.definition.model.flow.Flow;
+import io.gravitee.definition.model.flow.Operator;
+import io.gravitee.gateway.jupiter.api.context.Request;
+import io.gravitee.gateway.jupiter.api.context.RequestExecutionContext;
+import io.reactivex.observers.TestObserver;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@ExtendWith(MockitoExtension.class)
+class PathBasedConditionFilterTest {
+
+    @Mock
+    private RequestExecutionContext ctx;
+
+    @Mock
+    private Request request;
+
+    @Mock
+    private Flow flow;
+
+    private final PathBasedConditionFilter cut = new PathBasedConditionFilter();
+
+    @BeforeEach
+    void init() {
+        when(ctx.request()).thenReturn(request);
+    }
+
+    @Test
+    public void shouldFilterWhenPathNotEquals() {
+        when(request.pathInfo()).thenReturn("/my/path");
+        when(flow.getOperator()).thenReturn(Operator.EQUALS);
+        when(flow.getPath()).thenReturn("/my/path2");
+
+        final TestObserver<Flow> obs = cut.filter(ctx, flow).test();
+        obs.assertResult();
```

Comment:
not sure but maybe `assertEmpty` do the same and is more meaningful ?

Reply:
No it is not exactly the same, `assertResult` check that the observer completes with no value whereas `assertEmpty` check that the observer hasn't complete.